test/e2e/functional_test.go

```diff
@@ -65,6 +65,58 @@ func (s *FunctionalSuite) TestResourceQuota() {
 		})
 }
 
+// Test setting input artifact permission mode with and without the "recurseMode" boolean toggle
+func (s *FunctionalSuite) TestRecursiveArtifactMode() {
+	// Note: the "recurseMode" key is only used if it is set at the template input level rather than task
+	// argument. This behaviour is confirmed by the "artifact-mode-recurse-argument" step which shows
+	// that despite setting "recurseMode" to "true" in the task arguments the mode of the contents of
+	// the "message" artifact folder are not changed to "777".
+	s.Given().
```

Comment:
I think we might be better off with a unit test instead - is that possible please?

Reply:
No worries. Are you just wanting the test for the `LoadArtifacts` function?

Reply:
I always like to thing about what behaviour or feature is tested, rather than what software component is tested.

There, could the test make sure that recursive mode is applied when specified.